### PR TITLE
Initial Support for BouncyCastle FIPS

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR adds an initial support for BC in a FIPS mode, at the moment it is only verified that it is registered and can be used to load a BC specific signature instance; using it to secure 2-way TLS will be verified asap once we update to the newer versions of Vert.x.

### CODE REVIEW

File: bom/application/pom.xml
Comment (gastaldi) [inline bom/application/pom.xml:18]: Is this version somehow related to the `bouncycastle.version` above? If so, maybe it would be nice to add a comment to let maintainers know that this version needs to be updated if the version above changes

File: docs/src/main/asciidoc/security-customization.adoc
Comment (gastaldi) [inline docs/src/main/asciidoc/security-customization.adoc:287]: IMHO this step is a bit cryptic (no pun intended) :)

It seems easier to do have something like `quarkus.security.fips.enabled=true` than to remember to use `BCFIPS` here

File: bom/application/pom.xml
Comment (sberyozkin) [reply bom/application/pom.xml:18]: Hi @gastaldi No, `bc-fips` repackages a good number of classes from the regular BC provider and adds some add ons

File: docs/src/main/asciidoc/security-customization.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-customization.adoc:287]: @gastaldi users will have to remember either option or just check the docs :-)

`BCFIPS` is a well-known provider name known to anyone who wants to use Bouncy Castle and FIPS and it is consistent with how BouncyCastle itself recommends enabling BC + FIPS (i.e register a security provider named `BCFIPS`) and here we do the same, register this provider via `quarkus.security.security-providers`.

If we start with `quarkus.security.fips.enabled=true` then it will become ambiguous IMHO, should it work with the default `SunJSSE` which also has some FIPS code ? Or what does this mean, given the default providers section above:
```
quarkus.security.security-providers=SunRsaSign
quarkus.security.fips.enabled=true
```

So IMHO having an `atomic` `BCFIPS` is all right as we only plan to support FIPS in combination with BouncyCastle. Have I convinced you a little bit :-) ?

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-customization.adoc:287]: Then `BCFIPSJSSE` or similar will follow, a single documented 'convention' name, BC docs recommend setting 3 providers, and some properties, but in Quarkus it will be a one liner - enable BC + FIPS to secure 2-way TLS

Comment (gastaldi) [reply docs/src/main/asciidoc/security-customization.adoc:287]: Fair enough. I was under the impression that `quarkus.security.security-providers` would contain FQNs of `java.security.Provider` implementations

**Review by gastaldi — DISMISSED** [review]
LGTM, but there are some build errors

File: docs/src/main/asciidoc/security-customization.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-customization.adoc:287]: @gastaldi see, we are not going to support `fips` generally, that is the problem with yet another property IMHO. We have `security-providers` one and that should be enough to register a provider who can do whatever it can do, it is generic

**Comment by sberyozkin** [comment]
Hi @gastaldi Thanks, I won't rush with the merge anyway - I'll give it one more day from the moment you decide to approve to see and wait if @stuartwdouglas has something to comment upon :-) ; and sorry for putting too many arguments in favour of just `BCFIPS` :-)
It is a bit fluid right now, but I was thinking about it during my evening walk and it is indeed a cryptic name but all security provider names are cryptic :-) but the thing there is that they are not cryptic for people who need these providers :-), they know what they want to do with them, read the docs about them, but it is not a mainstream property, that is true :-)

Having said all that, once I get BC FIPS + JSSE working then I'll be happy to do another round of the discussions (how to ship it, how to configure it, etc). The messy part about FIPS is that we have `BCFIPS` but it can work with either default `SunJSSE` (though the docs do not really recommend it) and with the standard `BCJSSE` provider name. So I'm thinking right now if the provider name `BCFIPSJSSE` then we should check if BCJSSE is loaded, if yes - try it, if not - default to SunJSSE (as it is in this PR), I'll have to finalize once Vert.x is updated

Cheers

**Review by gastaldi — APPROVED** [review]

**Comment by sberyozkin** [comment]
Hi George @gastaldi I've resolved the conflict but also updated `.github/dependabot`  so that we can find out about the new BC releases asap

File: .github/dependabot.yml
Comment (gastaldi) [inline .github/dependabot.yml:124]: Maybe it's easier to just do:
```suggestion
      - dependency-name: org.bouncycastle:*
```

WDYT?

Comment (sberyozkin) [reply .github/dependabot.yml:124]: Hi @gastaldi I think we had some problems with a wildcard for some group id with a similar situation, here we `bc-fips` is on a different version `1.0.2` to the other 2 deps (`1.66`) ? I recall we discussed something similar with Guillaume, CC @gsmet

Comment (gastaldi) [reply .github/dependabot.yml:124]: Ah okay, no problem then 👍

**Pull request merged**
